chapters/ch06.md

```diff
@@ -1,48 +1,48 @@
-
+Nhưng nếu tập phát triển và tập kiểm tra được tạo từ các tập phân phối khác nhau thì việc xác định vấn đề sẽ phức tạp hơn. Rất nhiều thứ có thể sai:   
+  
 > 1. You had overfit to the dev set.
 
-1. Bạn đã overfit tập phát triển.
+1. Tập phát triển của bạn bị thiếu dữ liệu.
 
 > 2. The test set is harder than the dev set. So your algorithm might be doing as well as could be expected, and no further significant improvement is possible.
 
-2. Dữ liệu tập kiểm tra đa dạng (phức tạp) hơn tập phát triển. Thuật toán của bạn có thể đã hoạt động tốt trên tập phát triển, và không thể có thêm cải thiện đáng kể.
+2. Dữ liệu tập kiểm tra khó phân tích hơn tập phát triển. Vì vậy thuật toán của bạn có thể xem như đã hoạt động tốt như mong đợi, và không có cải thiện đáng kể nào khác được thực hiện.
 
 > 3. The test set is not necessarily harder, but just different, from the dev set. So what works well on the dev set just does not work well on the test set. In this case, a lot of your work to improve dev set performance might be wasted effort.
 
-3. Tập kiểm tra có thể không khó hơn nhưng lại khác so với tập phát triển. Do đó mô hình có thể hoạt động tốt trên tập phát triển nhưng lại kém trên tập kiểm tra. Trong trường hợp này, việc cố gắng cải thiện mô hình để hoạt động tốt trên tập phát triển có thể trở nên vô nghĩa.
+3. Tập kiểm tra có thể không khó hơn nhưng lại khác so với tập phát triển. Do đó những gì hoạt động tốt trên tập phát triển lại kém trên tập kiểm tra. Trong trường hợp này, nhiều cố gắng cải tiến hoạt động trên tập phát triển có thể trở nên lãng phí.
 
 > Working on machine learning applications is hard enough. Having mismatched dev and test sets introduces additional uncertainty about whether improving on the dev set distribution also improves test set performance. Having mismatched dev and test sets makes it harder to figure out what is and isn’t working, and thus makes it harder to prioritize what to work on.
 
-Nhiều vấn đề phức tạp có thể xảy ra khi làm việc với các ứng dụng machine learning. Tập phát triển và kiểm tra không nhất quán dẫn đến sự không chắc chắn về việc liệu rằng cải thiện hiệu quả trên tập phát triển có giúp cải thiện hiệu quả trên tập kiểm tra. Sự không nhất quán này khiến việc tìm ra nguyên nhân và sắp xếp hướng giải quyết trở nên khó khăn hơn vì không biết mô hình gặp vấn đề ở đâu. Tập phát triển và tập kiểm tra khác nhau cũng khiến cho việc xác định vấn đề trở nên khó khăn hơn, nhất là trong việc lựa chọn các công việc ưu tiên cần thiết để cải thiện mô hình.
+Nhiều vấn đề phức tạp có thể xảy ra khi làm việc với các ứng dụng machine learning. Tập phát triển và tập kiểm tra không khớp nhau dẫn đến sự không chắc chắn về việc liệu rằng cải thiện tập phát triển có giúp cải thiện hiệu quả trên tập kiểm tra hay không. Sự không khớp giữa tập phát triển và tập kiểm tra khiến cho việc tìm ra cái gì đang hoạt động cái gì không trở nên khó khăn hơn, và vì thế khó biết cần ưu tiên làm cái nào trước hơn.
```

Comment:
> liệu rằng cải thiện tập phát triển có giúp cải thiện hiệu quả trên tập kiểm tra hay không

mình nghĩ là nên có "cải thiện hiệu quả trên tập phát triển", nếu bạn muốn tránh lặp thì có thể viết như sau: 
> liệu rằng cải thiện hiệu quả trên tập phát triển có giúp đạt được điều tương tự trên tập kiểm tra hay không

Reply:
Mình xem kỹ lại thì tác giả viết là "improve on distribution", tức là cải thiện phân phối chứ không phải "cải thiện hiệu quả" bạn à (hiệu quả là kết quả đạt được sau việc cải thiện này). Vì sao cải thiện phân phối của tập phát triển có thể giúp tập kiểm tra hoạt động tốt hơn thì mọi người hãy bàn thêm. Có thể ở đây tác giả dùng chữ "whether" nên cũng không chắc chắn.

Reply:
improve distribution = cải thiện phân phối; improve on dev set distribution = cải thiệm hiệu quả (phân loại) trên tập phát triển? trước cụm  này có nói dev set mismatched với test set, sau vế này có nói improve test set performance. Bạn double check hộ mình.

Reply:
Cùng phân tích cấu trúc câu nha bạn:
Having mismatched dev and test sets / introduces / additional uncertainty about whether improving on the dev set distribution also improves test set performance.

additional uncertainty about / whether improving on the dev set distribution also improves test set performance

whether improving on the dev set distribution / also improves test set performance

improving on / the dev set distribution

the dev set distribution là danh từ ghép, trong đó distribution là từ chính còn dev set là từ phụ

Reply:
bạn đọc qua bản dịch update ở đây chưa: https://github.com/aivivn/Machine-Learning-Yearning-Vietnamese-Translation/blob/master/chapters/ch06.md ?

Reply:
@freeloneman "improve on somthing" và "improve somthing" là hai khái niệm hoàn toàn khác nhau. Mình không nghĩ chúng ta nên lãng phí thời gian tranh luận thêm vấn đề này.